### DISCUSSION TITLE
Various complaints from security team about velero helm charts

### DISCUSSION THREAD

**Discussion by doman18** [vmware-tanzu/velero#5169] [Community support Q&A]
Hello everyone
Our project had security audit some time ago. They scanned our clusters with these tools. 

• STARBOARD (https://github.com/aquasecurity/starboard)
• LENS IDE (https://k8slens.dev)
• KUBESCAPE (https://github.com/armosec/kubescape)
• sKan (https://github.com/alcideio/skan)

Results of those scans showed some vulnerabilities and list of affected containers contain also velero.

1. High-risk host file system volumes were mounted:
- velero/restic-*

NOTE: probably they refer to these (but i have to confirm this):
    volumeMounts:
    - mountPath: /credentials
      name: cloud-credentials
    - mountPath: /host_pods
      mountPropagation: HostToContainer
      name: host-pods
    - mountPath: /scratch
      name: scratch
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-k5hhf
      readOnly: true

2. Base Docker images were allowed to run processes as root user:
- velero/restic-*
- velero/velero-*

3. Containers capabilities were not dropped (its about tools not needed in image for make it work, but dangerous - kill, setuid, chown etc):
- velero/restic-*
- velero/velero-*

4. Liveness and readiness probes not configured:
- velero/restic
- velero/velero

So my question is: what can be done with those complaints?

**Comment by nsagark** [comment]
I have the same set of questions, and we need to fix these as part of remediating the security violations. If anybody has remediated these issues, please let me know.
